### PR TITLE
doc: update Raft doc for versions 5.2 and 2023.1

### PR DESCRIPTION
**Pull request by annastuchlik**
Fixes https://github.com/scylladb/scylladb/issues/13345
Fixes https://github.com/scylladb/scylladb/issues/13421

This commit updates the Raft documentation page to be up to date in versions 5.2 and 2023.1.

- Irrelevant information about previous releases is removed.
- Some information is clarified.
- Mentions of version 5.2 are either removed (if possible) or version 2023.1 is added.

### CODE REVIEW

**Comment by annastuchlik** [comment]
@kbr-scylla Could you review this PR? It fixes two issues, but I added the changes in one commit, as they are basically about the same topic and need to be backported to branch-5.2.

File: docs/architecture/raft.rst [hunk 1]
```diff
@@ -26,74 +26,42 @@ Raft requires at least a quorum of nodes in a cluster to be available. If multip
...
 .. note::
-  In ScyllaDB 5.2, Raft is Generally Available and can be safely used for consistent schema management.
-  In ScyllaDB 5.3 it will become enabled by default.
-  In further versions it will be mandatory.
+  In ScyllaDB 5.2 and 2023.1 Raft is Generally Available and can be safely used for consistent schema management.
+  In the next ScyllaDB version, it will become enabled by default.
+  In further versions, it will be mandatory.
 
-ScyllaDB 5.2 and later comes equipped with a procedure that can setup Raft-based consistent cluster management in an existing cluster. We refer to this as the **internal Raft upgrade procedure** (do not confuse with the :doc:`ScyllaDB version upgrade procedure </upgrade/upgrade-opensource/upgrade-guide-from-5.1-to-5.2/upgrade-guide-from-5.1-to-5.2-generic>`).
+ScyllaDB 5.2 and later and 2023.1 and later come equipped with a procedure that can setup Raft-based consistent cluster management in an existing cluster. We refer to this as the **Raft upgrade procedure** (do not confuse with the :doc:`ScyllaDB version upgrade procedure </index>`).
```
Comment (kbr-scylla) [inline docs/architecture/raft.rst:43]: You removed the word `internal` from `internal Raft upgrade procedure` which is fine, but you didn't do it in other mentions of `internal Raft upgrade procedure`. Let's make it consistent.

File: docs/architecture/raft.rst [hunk 2]
```diff
@@ -26,74 +26,42 @@ Raft requires at least a quorum of nodes in a cluster to be available. If multip
...
-
-.. TODO include enterprise versions in this documentation
+Enabling Raft
+---------------
 
 .. note::
-  In ScyllaDB 5.2, Raft is Generally Available and can be safely used for consistent schema management.
-  In ScyllaDB 5.3 it will become enabled by default.
-  In further versions it will be mandatory.
+  In ScyllaDB 5.2 and 2023.1 Raft is Generally Available and can be safely used for consistent schema management.
```
Comment (tzach) [inline docs/architecture/raft.rst:39]:
```suggestion
  In ScyllaDB 5.2 and ScyllaDB Enterprise 2023.1 Raft is Generally Available and can be safely used for consistent schema management.
```

File: docs/architecture/raft.rst [hunk 3]
```diff
@@ -26,74 +26,42 @@ Raft requires at least a quorum of nodes in a cluster to be available. If multip
...
+  In the next ScyllaDB version, it will become enabled by default.
```
Comment (tzach) [inline docs/architecture/raft.rst:40]:
```suggestion
```

Better not commit to future updates in docs.
We can say "in a followup release" or drop

File: docs/architecture/raft.rst (hunk 1 above)
Comment (tzach) [inline docs/architecture/raft.rst:43]:
```suggestion
ScyllaDB Open Source 5.2 and later, and ScyllaDB Enterprise 2023.1 and later come equipped with a procedure that can setup Raft-based consistent cluster management in an existing cluster. We refer to this as the **Raft upgrade procedure** (do not confuse with the :doc:`ScyllaDB version upgrade procedure </index>`).
```

**Comment by annastuchlik** [comment]
I've messed up something after applying your feedback :( I'll open another PR that will include your suggestions.

**Pull request closed without merging**